wire/msgtx.go

```diff
@@ -366,6 +367,29 @@ func (msg *MsgTx) TxHash() chainhash.Hash {
 	return chainhash.DoubleHashH(buf.Bytes())
 }
 
+// TxIdFromMsgTxHash generates the transaction ID (txid) of the transaction from
+// the double hash of the transaction.
+func TxIdFromMsgTxHash(txHash chainhash.Hash) string {
+	// Reverse the bytes of the hash.
+	revBytes := txHash
+	size := chainhash.HashSize
+	for i := 0; i < size/2; i++ {
+		revBytes[i], revBytes[size-1-i] = revBytes[size-1-i], revBytes[i]
+	}
+
+	// Hex encode the reversed bytes.
+	return hex.EncodeToString(revBytes[:])
+}
+
+// TxID generates the transaction ID of the transaction.
+func (msg *MsgTx) TxID() string {
+	// Encode the transaction and calculate the double sha256 hash of the
```

Comment:
Why not just:

```
// TxID is a wrapper around TxHash and returns a string of the transaction hash.
func (msg *MsgTx) TxID() string {
        return msg.TxHash().String()
}
```

Reply:
Because sometimes we may already have the `TxHash` available. If we have a function for computing the txid from the txhash then we can skip recomputing the hash.

However, I've gone with your suggestion of `msg.TxHash().String()` because all I really care about right now is adding `MsgTx.TxID`.